File: lib/js/CVManager.js
const fs = require('fs');

/**
 * Reads a CV from a JSON file.
 * 
 * @param {string} filePath - The path to the JSON file.
 * @returns {Object} - An object representing the CV.
 * 
 * @example
 * const cv = readCV('path/to/your/file.json');
 */
function readCV(filePath) {
    const json = fs.readFileSync(filePath, 'utf8');
    return JSON.parse(json);
}

/**
 * Writes a CV to a JSON file.
 * 
 * @param {Object} cv - An object representing the CV.
 * @param {string} filePath - The path to the JSON file where the CV will be saved.
 * 
 * @example
 * writeCV(cv, 'path/to/new/file.json');
 */
function writeCV(cv, filePath) {
    const json = JSON.stringify(cv, null, 4);
    fs.writeFileSync(filePath, json, 'utf8');
}

/**
 * Example usage:
 * 
 * const cv = readCV('path/to/your/file.json');
 * writeCV(cv, 'path/to/new/file.json');
 */

File: lib/csharp/CVManager.cs
using Newtonsoft.Json;
using System;
using System.IO;

/// <summary>
/// Manages reading and writing CVs in JSON format.
/// </summary>
/// <example>
/// This sample shows how to use the <see cref="CVManager"/> class.
/// <code>
/// CVManager manager = new CVManager();
/// dynamic cv = manager.ReadCV("path/to/your/file.json");
/// manager.WriteCV(cv, "path/to/new/file.json");
/// </code>
/// </example>
public class CVManager
{
    /// <summary>
    /// Reads a CV from a JSON file.
    /// </summary>
    /// <param name="filePath">The path to the JSON file.</param>
    /// <returns>A dynamic object representing the CV.</returns>
    public dynamic ReadCV(string filePath)
    {
        string json = File.ReadAllText(filePath);
        dynamic cv = JsonConvert.DeserializeObject(json);
        return cv;
    }

    /// <summary>
    /// Writes a CV to a JSON file.
    /// </summary>
    /// <param name="cv">The dynamic object representing the CV.</param>
    /// <param name="filePath">The path to the JSON file where the CV will be saved.</param>
    public void WriteCV(dynamic cv, string filePath)
    {
        string json = JsonConvert.SerializeObject(cv, Formatting.Indented);
        File.WriteAllText(filePath, json);
    }
}

